### PR TITLE
Add support for symbol link inside docker tar file

### PR DESCRIPTION
**Pull request by yongxiu**
docker tar file containing multiple images will have symbol link for layer files to save space, this will read from the target file instead of the symbol link file

### CODE REVIEW

File: pkg/v1/tarball/image.go [hunk 1]
```diff
@@ -216,6 +218,10 @@ func extractFileFromTar(opener Opener, filePath string) (io.ReadCloser, error) {
 			return nil, err
 		}
 		if hdr.Name == filePath {
+			if hdr.Typeflag == tar.TypeSymlink || hdr.Typeflag == tar.TypeLink {
+				currentDir := filepath.Dir(filePath)
```
Comment (jonjohnsonjr) [inline pkg/v1/tarball/image.go:222]: I have one concern about using `filepath.Dir` here. I'm not familiar with e.g. windows, but would this work when running on a windows machine if the tarball has normal path separators?

I think it's fine based on [this](https://golang.org/src/os/path_windows.go#L12) but I'd love some confirmation.

**Review by jonjohnsonjr — COMMENTED** [review]
Overall lgtm other than that one question. Can you add a test as well?

There are some test tarball archives under https://github.com/google/go-containerregistry/tree/main/pkg/v1/tarball/testdata

File: pkg/v1/tarball/image.go (hunk 1 above)
Comment (yongxiu) [reply pkg/v1/tarball/image.go:222]: I just tested in Windows, works same.

**Comment by yongxiu** [comment]
> Overall lgtm other than that one question. Can you add a test as well?
> 
> There are some test tarball archives under https://github.com/google/go-containerregistry/tree/main/pkg/v1/tarball/testdata

Added, please take another look, thanks.

**Review by jonjohnsonjr — APPROVED** [review]

**Comment by jonjohnsonjr** [comment]
Thanks!

**Pull request merged**